lib/IDE/SourceEntityWalker.cpp

```diff
@@ -236,6 +236,22 @@ ASTWalker::PreWalkAction SemaAnnotator::walkToDeclPreProper(Decl *D) {
 ASTWalker::PostWalkAction SemaAnnotator::walkToDeclPost(Decl *D) {
   auto Action = walkToDeclPostProper(D);
   SEWalker.endBalancedASTOrderDeclVisit(D);
+
+  if (Action.Action == PostWalkAction::Stop)
+    return Action;
+
+  // Walk into peer and conformance expansions if walking expansions
+  if (shouldWalkMacroArgumentsAndExpansion().second) {
+    D->visitAuxiliaryDecls([&](Decl *auxDecl) {
+      if (Action.Action == PostWalkAction::Stop)
```

Comment:
Out of curiosity, I wonder how bad it would be if the ASTWalker did this by default when visiting a Decl? Does seem kind of weird that it will visit some expansions but not others when you ask it to walk expansions

Reply:
I originally put it in Decl::walk (https://github.com/apple/swift/pull/66399). That's incorrect since `doIt` is called directly in the walk and thus we'd miss visiting auxiliary decls on members (note the test I added since then).

I could have moved to eg. doIt(Decl *) but there's callers that really expect to just walk the given node (eg. TypeCheckFunctionBodyRequest calls performAbstractFuncDeclDiagnostics), so that also doesn't work. So I went with the easier change of only impacting SourceEntityWalker, which only has a couple uses that walk expansions.

Also odd is that we walk into the extension while possible within a type, rather than at the top level.

Reply:
Hrm yeah, that's fair